### ISSUE TITLE
[2.1.4] Cannot use custom key database path -> no keys loaded

### ISSUE THREAD

**Issue by EHJ-52n** [saturneric/GpgFrontend#172]
I am staring the app image `GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage`. The crash of #165 is fixed, but the application doesn't find any keys in both of my key database paths: `~/.gnupg`, `~/.gnupg_private/` when using the "Use Custom GnuPG Key Database Path" feature.

<details>
<summary>Log output</summary>

```
~/opt/gpg-frontend ./GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage            
[20240805 11:01:50.916] [core] [W] [15082] - integrated module directory at path:  "/tmp/.mount_GpgFroxABUAo/usr/modules"  not found, abort...
[20240805 11:01:51.276] [core] [W] [15073] - data object not found from disk, key:  "main_window_state"
[20240805 11:01:51.276] [core] [W] [15073] - data object not found from disk, key:  "general_settings_state"
[20240805 11:01:51.330] [core] [W] [15073] - data object not found from disk, key:  "__cache_manage_data_register_key_list"
[20240805 11:01:51.330] [core] [W] [15073] - data object not found from disk, key:  "__cache_data_favourite_key_pair"
[20240805 11:01:51.355] [core] [W] [15073] - data object not found from disk, key:  "key_server"
[20240805 11:01:51.357] [core] [W] [15073] - data object not found from disk, key:  "__cache_data_editor_unsaved_pages"
[20240805 11:01:56.304] [core] [W] [15073] - data object not found from disk, key:  "GnuPGControllerDialog_dialog_state"
~/opt/gpg-frontend [20240805 11:02:06.124] [core] [W] [15157] - integrated module directory at path:  "/tmp/.mount_GpgFrooqRPEj/usr/modules"  not found, abort...
[20240805 11:02:54.994] [core] [W] [15234] - integrated module directory at path:  "/tmp/.mount_GpgFro2UjMke/usr/modules"  not found, abort...
~/opt/gpg-frontend
```

</details>

<details>
<summary>Log output with "debug" log lovel</summary>

```
~/opt/gpg-frontend ./GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage --log-level debug
[20240805 11:09:54.984] [core] [D] [15689] - a new global singleton storage collection created, address: 0x4baec0d28b0
[20240805 11:09:54.984] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:54.985] [core] [I] [15689] - app path:  "/tmp/.mount_GpgFroXUx1yX/usr/bin"
[20240805 11:09:54.985] [core] [I] [15689] - app working path:  "/home/user/opt/gpg-frontend"
[20240805 11:09:54.985] [core] [I] [15689] - app data path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage"
[20240805 11:09:54.985] [core] [I] [15689] - app log path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage/logs"
[20240805 11:09:54.985] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec4e3cf0
[20240805 11:09:54.985] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec4e3cf0
[20240805 11:09:54.985] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec4e3cf0
[20240805 11:09:54.986] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:54.986] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec217080
[20240805 11:09:54.986] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec217080
[20240805 11:09:54.986] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec217080
[20240805 11:09:54.986] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:54.987] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec4eac20
[20240805 11:09:54.987] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec4eac20
[20240805 11:09:54.987] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec4eac20
[20240805 11:09:54.987] [core] [W] [15697] - integrated module directory at path:  "/tmp/.mount_GpgFroXUx1yX/usr/modules"  not found, abort...
[20240805 11:09:54.987] [core] [D] [15689] - dear module manager, is all module registered? answer:  false
[20240805 11:09:54.993] [core] [D] [15697] - all modules are loaded into memory.
[20240805 11:09:54.994] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:54.994] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec4d88b0
[20240805 11:09:54.994] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec4d88b0
[20240805 11:09:54.994] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec4d88b0
[20240805 11:09:55.000] [core] [I] [15689] - detected gpgconf path:  "/usr/bin/gpgconf"
[20240805 11:09:55.000] [core] [I] [15689] - detected gnupg path:  "/usr/bin/gpg"
[20240805 11:09:55.035] [default] [I] [15689] - current system default locale:  "en_US"
[20240805 11:09:55.035] [core] [D] [15713] - set custom gpgconf path:  "/usr/bin/gpg"
[20240805 11:09:55.035] [default] [I] [15689] - locale settings from config:  ""
[20240805 11:09:55.035] [default] [I] [15689] - application's target locale:  "en_US"
[20240805 11:09:55.036] [default] [D] [15689] - locale info: 
[20240805 11:09:55.036] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.040] [core] [D] [15713] - gpg context: offline mode: 0
[20240805 11:09:55.040] [core] [D] [15713] - gpg context: auto import missing key: 0
[20240805 11:09:55.040] [core] [D] [15713] - got gpgme version version from rt:  "1.22.0"
[20240805 11:09:55.040] [ui] [D] [15689] - ui init is done, but cor doesn't, going to waiting for core...
[20240805 11:09:55.040] [core] [D] [15713] - ctx set engine info, db path:  "/home/user/.gnupg" , app path:  "/usr/bin/gpg"
[20240805 11:09:55.040] [ui] [D] [15689] - need to waiting for env checking process
[20240805 11:09:55.046] [core] [D] [15713] - set custom gpgconf path:  "/usr/bin/gpg"
[20240805 11:09:55.052] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.052] [core] [D] [15713] - gpg context: offline mode: 0
[20240805 11:09:55.052] [core] [D] [15713] - gpg context: auto import missing key: 0
[20240805 11:09:55.052] [core] [D] [15713] - got gpgme version version from rt:  "1.22.0"
[20240805 11:09:55.052] [core] [D] [15713] - ctx set engine info, db path:  "/home/user/.gnupg" , app path:  "/usr/bin/gpg"
[20240805 11:09:55.058] [core] [D] [15713] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baf6050150
[20240805 11:09:55.058] [core] [D] [15713] - register channel object to instances map, channel: 0, address: 0x4baf6050150
[20240805 11:09:55.058] [core] [D] [15713] - set channel: 0 success, current channel object address: 0x4baf6050150
[20240805 11:09:55.058] [core] [D] [15713] - cannot find channel object, channel: 0
[20240805 11:09:55.058] [core] [D] [15713] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baf60517a0
[20240805 11:09:55.058] [core] [D] [15713] - register channel object to instances map, channel: 0, address: 0x4baf60517a0
[20240805 11:09:55.058] [core] [D] [15713] - set channel: 0 success, current channel object address: 0x4baf60517a0
[20240805 11:09:55.067] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.082] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.097] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.112] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.125] [ui] [D] [15689] - ui is ready to waiting for env initialized, env_state: 0
[20240805 11:09:55.128] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.143] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.158] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.173] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.188] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.204] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.220] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.235] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.250] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.265] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.280] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.296] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.311] [core] [D] [15714] - monitor: core env is still initializing, waiting...
[20240805 11:09:55.325] [core] [D] [15713] - basic env checking finished, including gpgme, ctx, and key infos
[20240805 11:09:55.326] [core] [D] [15714] - module manager report, need register:  0 registered 0
[20240805 11:09:55.326] [core] [D] [15714] - monitor: good, all module are registered.
[20240805 11:09:55.326] [core] [D] [15714] - monitor: core is fully initialized, sending signal to ui...
[20240805 11:09:55.326] [ui] [D] [15689] - ui caught signal: core fully loaded
[20240805 11:09:55.326] [core] [D] [15689] - monitor task ended, call back to main thead.
[20240805 11:09:55.326] [ui] [D] [15689] - application fully initialized...
[20240805 11:09:55.330] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:55.331] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec423eb0
[20240805 11:09:55.331] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec423eb0
[20240805 11:09:55.331] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec423eb0
[20240805 11:09:55.334] [ui] [D] [15689] - main window start to initialize...
[20240805 11:09:55.385] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:09:55.385] [core] [D] [15689] - start to load all cache from file system
[20240805 11:09:55.385] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec61dee0
[20240805 11:09:55.385] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec61dee0
[20240805 11:09:55.385] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec61dee0
[20240805 11:09:55.412] [core] [I] [15697] - event:  "APPLICATION_LOADED"  is not listening by anyone and not registered as well.
[20240805 11:10:09.946] [default] [W] [15689] - QFont::fromString: Invalid description '(empty)'
[20240805 11:10:09.954] [default] [W] [15689] - QFont::fromString: Invalid description '(empty)'
[20240805 11:10:10.177] [default] [W] [15689] - QFont::fromString: Invalid description '(empty)'
[20240805 11:10:10.265] [core] [D] [15689] - write cache to file system...
[20240805 11:10:15.540] [default] [W] [15689] - QFont::fromString: Invalid description '(empty)'
[20240805 11:10:15.540] [default] [W] [15689] - QFont::fromString: Invalid description '(empty)'
[20240805 11:10:25.216] [core] [D] [15689] - write cache to file system...
[20240805 11:10:28.187] [core] [D] [15689] - global singleton storage collection is about to destroy, address: 0x4baec0d28b0
[20240805 11:10:28.187] [core] [D] [15689] - write cache to file system...
[20240805 11:10:28.188] [core] [D] [15689] - a new global singleton storage collection created, address: 0x4baec0d93a0
[20240805 11:10:28.188] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:10:28.188] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:10:28.188] [core] [I] [15689] - app path:  "/tmp/.mount_GpgFroXUx1yX/usr/bin"
[20240805 11:10:28.188] [core] [I] [15689] - app working path:  "/home/user/opt/gpg-frontend"
[20240805 11:10:28.188] [core] [I] [15689] - app data path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage"
[20240805 11:10:28.188] [core] [I] [15689] - app log path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage/logs"
[20240805 11:10:28.188] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baed942e90
[20240805 11:10:28.188] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baed942e90
[20240805 11:10:28.188] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baed942e90
[20240805 11:10:28.188] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baed96a120
[20240805 11:10:28.188] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baed96a120
[20240805 11:10:28.188] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baed96a120
[20240805 11:10:28.197] [core] [D] [15689] - cannot find channel object, channel: 0
[20240805 11:10:28.197] [core] [D] [15689] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x4baec6f7900
[20240805 11:10:28.197] [core] [D] [15689] - register channel object to instances map, channel: 0, address: 0x4baec6f7900
[20240805 11:10:28.197] [core] [D] [15689] - set channel: 0 success, current channel object address: 0x4baec6f7900
[20240805 11:10:28.197] [core] [D] [15689] - global singleton storage collection is about to destroy, address: 0x4baec0d93a0
~/opt/gpg-frontend [20240805 11:10:28.398] [core] [D] [15775] - a new global singleton storage collection created, address: 0x57e580d2240
[20240805 11:10:28.399] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.399] [core] [I] [15775] - app path:  "/tmp/.mount_GpgFroUG9PtB/usr/bin"
[20240805 11:10:28.399] [core] [I] [15775] - app working path:  "/home/user/opt/gpg-frontend"
[20240805 11:10:28.399] [core] [I] [15775] - app data path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage"
[20240805 11:10:28.399] [core] [I] [15775] - app log path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage/logs"
[20240805 11:10:28.399] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e584e7bc0
[20240805 11:10:28.399] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e584e7bc0
[20240805 11:10:28.399] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e584e7bc0
[20240805 11:10:28.400] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.400] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e58213c00
[20240805 11:10:28.400] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e58213c00
[20240805 11:10:28.400] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e58213c00
[20240805 11:10:28.400] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.401] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e584e6f90
[20240805 11:10:28.401] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e584e6f90
[20240805 11:10:28.401] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e584e6f90
[20240805 11:10:28.401] [core] [D] [15775] - dear module manager, is all module registered? answer:  false
[20240805 11:10:28.401] [core] [W] [15795] - integrated module directory at path:  "/tmp/.mount_GpgFroUG9PtB/usr/modules"  not found, abort...
[20240805 11:10:28.408] [core] [D] [15795] - all modules are loaded into memory.
[20240805 11:10:28.408] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.408] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e584db620
[20240805 11:10:28.408] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e584db620
[20240805 11:10:28.408] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e584db620
[20240805 11:10:28.414] [core] [I] [15775] - detected gpgconf path:  "/usr/bin/gpgconf"
[20240805 11:10:28.414] [core] [I] [15775] - detected gnupg path:  "/usr/bin/gpg"
[20240805 11:10:28.450] [default] [I] [15775] - current system default locale:  "en_US"
[20240805 11:10:28.450] [default] [I] [15775] - locale settings from config:  ""
[20240805 11:10:28.450] [default] [I] [15775] - application's target locale:  "en_US"
[20240805 11:10:28.450] [default] [D] [15775] - locale info: 
[20240805 11:10:28.450] [core] [D] [15812] - monitor: core env is still initializing, waiting...
[20240805 11:10:28.450] [core] [D] [15811] - set custom gpgconf path:  "/usr/bin/gpg"
[20240805 11:10:28.454] [ui] [D] [15775] - ui init is done, but cor doesn't, going to waiting for core...
[20240805 11:10:28.454] [ui] [D] [15775] - need to waiting for env checking process
[20240805 11:10:28.456] [core] [D] [15811] - gpg context: offline mode: 0
[20240805 11:10:28.456] [core] [D] [15811] - gpg context: auto import missing key: 0
[20240805 11:10:28.456] [core] [D] [15811] - got gpgme version version from rt:  "1.22.0"
[20240805 11:10:28.456] [core] [D] [15811] - ctx set engine info, db path:  "/usr/bin" , app path:  "/usr/bin/gpg"
[20240805 11:10:28.462] [core] [D] [15811] - set custom gpgconf path:  "/usr/bin/gpg"
[20240805 11:10:28.466] [core] [D] [15812] - monitor: core env is still initializing, waiting...
[20240805 11:10:28.467] [core] [D] [15811] - gpg context: offline mode: 0
[20240805 11:10:28.467] [core] [D] [15811] - gpg context: auto import missing key: 0
[20240805 11:10:28.467] [core] [D] [15811] - got gpgme version version from rt:  "1.22.0"
[20240805 11:10:28.468] [core] [D] [15811] - ctx set engine info, db path:  "/usr/bin" , app path:  "/usr/bin/gpg"
[20240805 11:10:28.473] [core] [D] [15811] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e620500c0
[20240805 11:10:28.473] [core] [D] [15811] - register channel object to instances map, channel: 0, address: 0x57e620500c0
[20240805 11:10:28.473] [core] [D] [15811] - set channel: 0 success, current channel object address: 0x57e620500c0
[20240805 11:10:28.473] [core] [D] [15811] - cannot find channel object, channel: 0
[20240805 11:10:28.473] [core] [D] [15811] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e620506c0
[20240805 11:10:28.473] [core] [D] [15811] - register channel object to instances map, channel: 0, address: 0x57e620506c0
[20240805 11:10:28.473] [core] [D] [15811] - set channel: 0 success, current channel object address: 0x57e620506c0
[20240805 11:10:28.480] [core] [D] [15811] - basic env checking finished, including gpgme, ctx, and key infos
[20240805 11:10:28.481] [core] [D] [15812] - module manager report, need register:  0 registered 0
[20240805 11:10:28.481] [core] [D] [15812] - monitor: good, all module are registered.
[20240805 11:10:28.481] [core] [D] [15812] - monitor: core is fully initialized, sending signal to ui...
[20240805 11:10:28.540] [ui] [D] [15775] - ui is ready to waiting for env initialized, env_state: 1
[20240805 11:10:28.540] [ui] [D] [15775] - env state turned initialized before the looper start
[20240805 11:10:28.540] [ui] [D] [15775] - application fully initialized...
[20240805 11:10:28.543] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.544] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e5876b020
[20240805 11:10:28.544] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e5876b020
[20240805 11:10:28.544] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e5876b020
[20240805 11:10:28.547] [ui] [D] [15775] - main window start to initialize...
[20240805 11:10:28.607] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:28.607] [core] [D] [15775] - start to load all cache from file system
[20240805 11:10:28.608] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e588d4710
[20240805 11:10:28.608] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e588d4710
[20240805 11:10:28.608] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e588d4710
[20240805 11:10:28.646] [core] [I] [15795] - event:  "APPLICATION_LOADED"  is not listening by anyone and not registered as well.
[20240805 11:10:28.742] [core] [D] [15775] - monitor task ended, call back to main thead.
[20240805 11:10:34.943] [core] [D] [15775] - write cache to file system...
[20240805 11:10:34.947] [core] [D] [15775] - global singleton storage collection is about to destroy, address: 0x57e580d2240
[20240805 11:10:34.947] [core] [D] [15775] - write cache to file system...
[20240805 11:10:34.947] [core] [D] [15775] - a new global singleton storage collection created, address: 0x57e580d8470
[20240805 11:10:34.947] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:34.947] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:34.947] [core] [I] [15775] - app path:  "/tmp/.mount_GpgFroUG9PtB/usr/bin"
[20240805 11:10:34.948] [core] [I] [15775] - app working path:  "/home/user/opt/gpg-frontend"
[20240805 11:10:34.948] [core] [I] [15775] - app data path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage"
[20240805 11:10:34.948] [core] [I] [15775] - app log path:  "/home/user/.local/share/GpgFrontend-2.1.4-qt5-linux-x86_64.AppImage/logs"
[20240805 11:10:34.948] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e592b9640
[20240805 11:10:34.948] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e592b9640
[20240805 11:10:34.948] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e592b9640
[20240805 11:10:34.948] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e58617200
[20240805 11:10:34.948] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e58617200
[20240805 11:10:34.948] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e58617200
[20240805 11:10:34.948] [core] [D] [15775] - cannot find channel object, channel: 0
[20240805 11:10:34.948] [core] [D] [15775] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x57e5893d580
[20240805 11:10:34.948] [core] [D] [15775] - register channel object to instances map, channel: 0, address: 0x57e5893d580
[20240805 11:10:34.948] [core] [D] [15775] - set channel: 0 success, current channel object address: 0x57e5893d580
[20240805 11:10:34.948] [core] [D] [15775] - global singleton storage collection is about to destroy, address: 0x57e580d8470

~/opt/gpg-frontend
``` 

</details>

I need to disable the "Use Custom GnuPG Key Database Path", so the application is finding gpg keys again.

Hence, the crash is "fixed", but the feature is gone.

Is there anything, I can do, to help fixing this issue?

Do you need any additional information from my side?

**Comment by saturneric** [comment]
Dieses Problem wurde durch einen Programmierfehler meinerseits verursacht. Die vom Benutzer eingestellten Parameter werden nicht wirklich auf den Kontext angewendet.

Refer: d4e2335bfdff0985d4e410bf608bf73ff1fcfca7

**Comment by EHJ-52n** [comment]
:+1: Thank you very much for your fast reaction! :partying_face: :tada: 

Looking forward to 2.1.5-gt5 with updated notifcation settings. :wink:

**Comment by saturneric** [comment]
The good news is that you don't have to wait at this stage. v2.1.4.3 has been released (you've just been using v2.1.4.2), fixing a number of software bugs that I've recently gathered. You can download it now and continue to use it or test it.

**Comment by EHJ-52n** [comment]
The error seems fixed with the latest version v2.1.4.3. The keys of both database paths are shown and switching is possible.
